src/lcm/entry_point.py

```diff
@@ -117,9 +138,119 @@ def get_lcm_function(model, targets="solve", interpolation_options=None):
         state_choice_spaces=state_choice_spaces,
         state_indexers=state_indexers,
         continuous_choice_grids=continuous_choice_grids,
-        utility_and_feasibility_functions=utility_and_feasibility_functions,
+        compute_ccv_functions=compute_ccv_functions,
         emax_calculators=emax_calculators,
-        choice_segments=choice_segments,
     )
 
-    return solve_model, _mod.params
+    next_state = get_next_state_function(model=_mod)
+
+    simulate_model = partial(
+        simulate,
+        state_indexers=state_indexers,
+        continuous_choice_grids=continuous_choice_grids,
+        compute_ccv_argmax_functions=compute_ccv_argmax_functions,
+        model=_mod,
+        next_state=next_state,
+    )
+
+    _target = solve_model if targets == "solve" else simulate_model
+    return _target, _mod.params
+
+
+def create_compute_conditional_continuation_value(
+    utility_and_feasibility,
+    continuous_choice_variables,
+):
+    """Create a function that computes the conditional continuation value.
+
+    Note:
+    -----
+    This function solves the continuous choice problem conditional on a state-
+    (discrete-)choice combination.
+
+    Args:
+        utility_and_feasibility (callable): A function that takes a state-choice
+            combination and return the utility of that combination (float) and whether
+            the state-choice combination is feasible (bool).
+        continuous_choice_variables (list): List of choice variable names that are
+            continuous.
+
+    Returns:
+        callable: A function that takes a state-choice combination and returns the
+            conditional continuation value over the continuous choices.
+
+    """
+    u_and_f_mapped_over_cont_choices = productmap(
+        func=utility_and_feasibility,
+        variables=continuous_choice_variables,
+    )
+
+    @functools.wraps(u_and_f_mapped_over_cont_choices)
+    def compute_ccv(*args, **kwargs):
+        u, f = u_and_f_mapped_over_cont_choices(*args, **kwargs)
+        return u.max(where=f, initial=-jnp.inf)
+
+    return compute_ccv
+
+
+def create_compute_conditional_continuation_value_argmax(
```

Comment:
If I see it correctly, this function is only tested via a test of the simulate function. I think we should test it in isolation!